code/src/components/TaskList.js

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable linebreak-style */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable linebreak-style */
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { tasks } from 'reducers/tasks';
+
+const TaskList = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const listfromStorage = JSON.parse(localStorage.getItem('allTasks'));
+    if (listfromStorage) {
+      dispatch(tasks.actions.setAllItems(listfromStorage));
+    }
+  }, [])
+  const allTasks = useSelector((store) => store.tasks.items);
+
+  const onIsTaskCompleted = (id) => {
+    dispatch(tasks.actions.toggleItem(id));
+  }
+  const onDeleteTaskBtnClick = (taskIndex) => {
+    dispatch(tasks.actions.deleteItem(taskIndex));
+  }
+  const functionForMap = (todo, broom) => {
+    return (
+
+      <section className="tasklist-container" key={todo.id}>
+        <div className="checkbox-container">
+          <p>{todo.text}</p>
+          <label className="button-container">
+            <input
+              type="checkbox"
+              id="task"
+              checked={todo.taskDone}
+              onChange={() => onIsTaskCompleted(todo.id)} />
+            <span className="checkmark" />
+          </label>
+        </div>
+        <div className="delete-container">
+          <button
+            className="broom-btn"
+            onClick={() => onDeleteTaskBtnClick(broom)}
+            type="button">🧹
+          </button>
+        </div>
+
+      </section>
+
+    );
+  }
+  return (
+    <section>
+      {allTasks.map((task, index) => functionForMap(task, index))}
+    </section>
```

Comment:
I have tried to look at the code here to figure out how to solve the ticking of the checkboxes... I have a totally different solution to it so it was hard to get it interpreted to yours and I haven't used functionFormap. Maybe there could be something with index and broom, maybe only to use index? But then I saw that you have the allTasks.map below the function. Maybe you can try to put the return on line 54 to line 29, I'm not sure if that works, but then you get the mapping inside the function.